### ISSUE TITLE
Cancel the two-factor authentication

### ISSUE THREAD

**Issue by xchs** [contao/contao#292]
**Affected version(s)**

Contao 4.6 – 4.7

**Description**

The route `contao/logout` for the 2FA **cancel** operation does not switch back to the Contao back end login screen. The page simply reloads. So you can't cancel the authentication process.

![2fa_cancel](https://user-images.githubusercontent.com/1218766/51429209-bf872080-1c0c-11e9-9a01-bca3a137d4d8.png)

**How to reproduce**

* Enable the two-factor authentication
* Log into the Contao back end with the user's credentials
* Try to cancel the two-factor authentication

**Comment by mktcode** [comment]
What's the point of cancelling that action anyway?

**Comment by mktcode** [comment]
Maybe this has something todo with the different token (`TwoFactorToken`) that is set in that step...

![image](https://user-images.githubusercontent.com/6792578/51465030-3f290280-1d67-11e9-8857-1c9359ec6822.png)

I'm not sure but I see code like
```
if (!$user instanceof User) {
    return;
}
```
in different places. This check would be true for that `TwoFactorToken` and just return. Maybe it has something to do with the issue.

**Comment by leofeyer** [comment]
It works with  `symfony/symfony` and `symfony/security-bundle` in version 4.1.3 and it no longer works with version 4.1.4, therefore one of the following changes must cause the issue:

- https://github.com/symfony/security/compare/v4.1.3...v4.1.4
- https://github.com/symfony/security-bundle/compare/v4.1.3...v4.1.4

I assume it is this one: https://github.com/symfony/symfony/pull/28100

**Comment by bytehead** [comment]
Adding this line to the security config works in my demo installation:
```yml
security:
    access_control:
    	- { path: ^/contao/logout, roles: IS_AUTHENTICATED_ANONYMOUSLY }
```

**Comment by leofeyer** [comment]
That is true, but most likely not the correct fix. The logout path has been exempt from `access_control` in Symfony ever since and even though this behavior was accidentally changed in https://github.com/symfony/symfony/pull/24805, it has been restored in https://github.com/symfony/symfony/pull/28100 (also see the discussion in https://github.com/symfony/symfony/issues/28058).

**Comment by bytehead** [comment]
You're right. Need to dig deeper.

**Comment by bytehead** [comment]
After some hours of debugging I've ended up with the following: to use `access_control` for the logout path seems to be the correct solution. It's also used in the test application for the two-factor-bundle: 
https://github.com/scheb/two-factor-app/blob/863c5d153c4780e9be77e84484207bb04933eef6/config/packages/security.yaml#L31-L33

**Comment by leofeyer** [comment]
Yeah, Christian also mentions it here: https://github.com/scheb/two-factor-bundle/issues/183#issuecomment-441702428 Thank you for your help anyway! 🙏

**Comment by xchs** [comment]
Thanks a lot for all the time spent debugging the issue. 👍

**Comment by leofeyer** [comment]
Fixed in 29e9eadb72cb14c02aa9ce4407f9d3357b1614f0.
